aten/src/ATen/native/mps/operations/BinaryOps.mm

```diff
@@ -27,6 +27,10 @@
 void binaryOpTensor(const Tensor& self, const Tensor& other, const Scalar& alpha,
                     const Tensor& output_, std::string op_name, BinaryOpBlock binaryBlock)
 {
+
+  TORCH_CHECK(!(self.scalar_type() == ScalarType::Long &&
```

Comment:
This code should be moved to `CREATE_MPS_STRUCTURED_BINARY_OP_FUNC()` function macro, because it's only used for power and atan2. We don't need this check for comparison ops (like eq, ne, etc.).

Reply:
Done